src/include/storage/storage_defs.h

```diff
@@ -82,7 +86,14 @@ struct alignas(common::Constants::BLOCK_SIZE) RawBlock {
   byte content_[common::Constants::BLOCK_SIZE - sizeof(uintptr_t) - sizeof(uint16_t) - sizeof(layout_version_t) -
                 sizeof(uint32_t) - sizeof(BlockAccessController)];
   // A Block needs to always be aligned to 1 MB, so we can get free bytes to
-  // store offsets within a block in ine 8-byte word.
+  // store offsets within a block in ine 8-byte word
+
+  /**
+   * Get the offset of this block. Because the first bit insert_head_ is used to indicate the status
+   * of the block, we need to clear the status bit to get the real offset
+   * @return the offset which tells us where the next insertion should take place
+   */
+  std::atomic<uint32_t> GetInsertHead() { return (~(1 << 31)) & insert_head_.load(); }
```

Comment:
Why is the return type also an atomic? This is not the same underlying value since you've copied it

Reply:
Will it cause any problem if I change the return type to uint32_t?
In other places, we use the offset to check if the block is full. They may use stale values to do the check?

Reply:
You are making a copy anyways so it will be stale even if it is atomic.

Reply:
I got that point. Any suggestions on how to fix it? I guess right now I will just leave some comments there because it passed all tests and seems not to affect too much.